lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -577,45 +583,58 @@ StoreInst *StackAllocationPromoter::promoteAllocationInBlock(
 
     if (isLoadFromStack(inst, asi)) {
       assert(!runningVals || runningVals->isStorageValid);
-      auto *li = cast<LoadInst>(inst);
-      if (li->getOwnershipQualifier() == LoadOwnershipQualifier::Take) {
-        if (shouldAddLexicalLifetime(asi)) {
-          // End the lexical lifetime at a load [take].  The storage is no
-          // longer keeping the value alive.
-          if (runningVals && canEndLexicalLifetime(*runningVals)) {
-            // End it right now if we have enough information.
-            endLexicalLifetimeBeforeInst(asi, /*beforeInstruction=*/li, ctx,
-                                         runningVals->value);
-          } else {
-            // If we dont't have enough information, end it endLexicalLifetime.
-            assert(!deinitializationPoints[blockPromotingWithin]);
-            deinitializationPoints[blockPromotingWithin] = li;
+      if (auto *li = dyn_cast<LoadInst>(inst)) {
```

Comment:
we're up to ~7 layers of indentation and if-blocks that span a couple pages. it's hard for me to know what condition I'm looking at. It seems like none of this needs to be indented if you have
```
  if (auto *lbi = cast<LoadBorrowInst>(inst)) {
    // a few lines of new code...
    continue;
  }
  // old code unchanged
```
...or any number of ways to better compose this code.

Reply:
Yep, diff looks much simpler now. thanks